app/src/main/java/com/brandon/forzenbook/viewmodels/LoginViewModel.kt

```diff
@@ -0,0 +1,239 @@
+                        CreateUserOutcome.CREATE_USER_FAILURE -> {
+                            CreateAccountUiState.Idle(
+                                email = email,
+                                firstName = firstName,
+                                lastName = lastName,
+                                location = location,
+                                dateOfBirth = date,
+                            )
+                        }
+                        CreateUserOutcome.CREATE_USER_DUPLICATE -> {
+                            CreateAccountUiState.Idle(
+                                email = email,
+                                firstName = firstName,
+                                lastName = lastName,
+                                location = location,
+                                dateOfBirth = date,
+                                isDuplicateUser = true,
+                            )
+                        }
+                        CreateUserOutcome.CREATE_USER_SUCCESS -> {
+                            CreateAccountUiState.Loaded
+                        }
+                    }
+                }
+            } else {
+                createAccountState.value = CreateAccountUiState.Idle(
+                    isEmailError = validMail,
+                    isLocationError = validLoc,
+                    isDateError = validDob,
+                    email = email,
+                    firstName = firstName,
+                    lastName = lastName,
+                    location = location,
+                    dateOfBirth = date,
+                )
+            }
+        }
+        Log.e(VIEWMODEL_ERROR_TAG, "${createAccountState.value}")
+    }
+
+    private fun isValidLocation(location: String): Boolean {
+        return location.length <= LOCATION_CHAR_LIMIT && location.isNotEmpty()
+    }
+
+    private fun isValidEmail(email: String): Boolean {
+        return PatternsCompat.EMAIL_ADDRESS.matcher(email).matches()
+    }
+
+    private fun isValidDateOfBirth(dob: String): Boolean {
+        val date = dob.filter { it != '/' && it != '-' }
+        date.forEach {
+            if (!it.isDigit()) {
+                return false
+            }
+        }
+        return try {
+            val inputFormat = SimpleDateFormat(INCOMING_DATE_FORMAT)
+            val tempDate = inputFormat.parse(date)
+            val outputFormat = SimpleDateFormat(OUTGOING_DATE_FORMAT)
+            val dateString = tempDate?.let { outputFormat.format(it) }
+            sqlDate = Date.valueOf(dateString)
+            true
+        } catch (e: Exception) {
+            Log.e(VIEWMODEL_ERROR_TAG, "Invalid Date of Birth Provided.")
+            false
+        }
+    }
+
+    private fun isValidCode(code: String): Boolean {
+        return code.isDigitsOnly() && code.length == CODE_CHAR_LIMIT
+    }
+
+    fun isNetworkAvailable(context: Context): Boolean {
+        val connectivityManager =
```

Comment:
Recreating this every time you call it is a massive performance hit.
Maybe inject application context and keep reference to it

Reply:
I have a solution for this, but want to run it by you first to see if its reasonable or a good approach.